cpp/test/sg/decisiontree_batchedlevel_algo.cu

```diff
@@ -0,0 +1,212 @@
+    auto alpha = T(1.0), beta = T(0.0);
+    auto cublas = handle->get_cublas_handle();
+    CUBLAS_CHECK(raft::linalg::cublasgeam(cublas,
+                                          CUBLAS_OP_T,
+                                          CUBLAS_OP_N,
+                                          inparams.M,
+                                          inparams.N,
+                                          &alpha,
+                                          tmp.data(),
+                                          inparams.N,
+                                          &beta,
+                                          tmp.data(),
+                                          inparams.M,
+                                          data->data(),
+                                          inparams.M,
+                                          stream));
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    rowids = std::make_unique<rmm::device_uvector<I>>(inparams.M, stream);
+    MLCommon::iota(rowids->data(), 0, 1, inparams.M, stream);
+    quantiles = std::make_unique<rmm::device_uvector<T>>(inparams.nbins * inparams.N, stream);
+
+    // computing the quantiles
+    computeQuantiles(
+      quantiles, inparams.nbins, data->data(), inparams.M, inparams.N, allocator, stream);
+  }
+
+  void TearDown() { CUDA_CHECK(cudaStreamDestroy(stream)); }
+
+  cudaStream_t stream;
+  std::shared_ptr<raft::handle_t> handle;
+  std::unique_ptr<rmm::device_uvector<T>> data, quantiles;
+  std::unique_ptr<rmm::device_uvector<L>> labels;
+  std::unique_ptr<rmm::device_uvector<I>> rowids;
+  DecisionTreeParams params;
+  DtTestParams inparams;
+  std::vector<SparseTreeNode<T, L>> sparsetree;
+
+  virtual void prepareDataset(T* tmp) = 0;
+};  // class DtBaseTest
+
+const std::vector<DtTestParams> allC = {
+  {1024, 4, 2, 8, 16, 0.00001f, CRITERION::GINI, 12345ULL},
+  {1024, 4, 2, 8, 16, 0.00001f, CRITERION::GINI, 12345ULL},
+  {1024, 4, 2, 8, 16, 0.00001f, CRITERION::ENTROPY, 12345ULL},
+  {1024, 4, 2, 8, 16, 0.00001f, CRITERION::ENTROPY, 12345ULL},
+};
+template <typename T>
+class DtClassifierTest : public DtBaseTest<T, int> {
+ protected:
+  void prepareDataset(T* tmp) override
+  {
+    auto inparams = this->inparams;
+    MLCommon::Random::make_blobs<T>(tmp,
+                                    labels->data(),
+                                    inparams.M,
+                                    inparams.N,
+                                    inparams.nclasses,
+                                    stream,
+                                    true,
+                                    nullptr,
+                                    nullptr,
+                                    T(1.0),
+                                    false,
+                                    T(10.0),
+                                    T(-10.0),
+                                    inparams.seed);
+  }
+};  // class DtClassifierTest
+typedef DtClassifierTest<float> DtClsTestF;
+///@todo: add checks
```

Comment:
Is this todo still relevant? If so, can you add more detail?

Reply:
No idea. This was in the code already.